src/plugin/declarativedbusabstractobject.h

```diff
@@ -0,0 +1,120 @@
+/****************************************************************************************
+**
+** Copyright (c) 2013 - 2021 Jolla Ltd.
+** All rights reserved.
+**
+** You may use this file under the terms of the GNU Lesser General
+** Public License version 2.1 as published by the Free Software Foundation
+** and appearing in the file license.lgpl included in the packaging
+** of this file.
+**
+** This library is free software; you can redistribute it and/or
+** modify it under the terms of the GNU Lesser General Public
+** License version 2.1 as published by the Free Software Foundation
+** and appearing in the file license.lgpl included in the packaging
+** of this file.
+**
+** This library is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+** Lesser General Public License for more details.
+**
+****************************************************************************************/
+
+#ifndef DECLARATIVEDBUSABSTRACTOBJECT_H
+#define DECLARATIVEDBUSABSTRACTOBJECT_H
+
+#include <QQmlListProperty>
+#include <QQmlParserStatus>
+#include <QUrl>
+#include <QJSValue>
+
+#include <QDBusVirtualObject>
+
+#include <QEventLoopLocker>
+
+#include "declarativedbus.h"
+
+class DeclarativeDBusAbstractObject : public QDBusVirtualObject, public QQmlParserStatus
+{
+    Q_OBJECT
+    Q_PROPERTY(QString service READ service WRITE setService NOTIFY serviceChanged)
+    Q_PROPERTY(QString path READ path WRITE setPath NOTIFY pathChanged)
+    Q_PROPERTY(QString xml READ xml WRITE setXml NOTIFY xmlChanged)
+    Q_PROPERTY(DeclarativeDBus::BusType bus READ bus WRITE setBus NOTIFY busChanged)
+    Q_PROPERTY(bool quitOnTimeout READ quitOnTimeout WRITE setQuitOnTimeout NOTIFY quitOnTimeoutChanged)
+    Q_PROPERTY(int quitTimeout READ quitTimeout WRITE setQuitTimeout NOTIFY quitTimeoutChanged)
```

Comment:
I have mixed feelings for these autoquit properties here. They do support d-bus related use-case, but then again the implementation side is not much related to d-bus adaptors, and the same property is available on two different components. Potentially later on third if we add a component to represent the d-bus service.

No clear home for the functionality, but pondering if a separate component in Silica could be justified. Perhaps not all auto-quit needs are related to d-bus.

Reply:
Even focusing on the really specific use case of exiting because a d-bus thing didn't happen or happened but didn't result in an application window being shown it's difficult to come up with something that isn't a foot gun.

It's here because I was looking for a way to cancel the quit after handling a method if there were visible windows or something, but that doesn't cover non-prestart, or the existance other dbus adaptors. But the other reason is that keeps expectations low, it does this one thing and so long as you follow the example your applications not going to exit because you got a phone call that coincided with an exit if not seemingly open use timeout. Something in silica that could be used for hypothetical non dbus auto start timeout use cases will come with an expectation that it will actually work for those use cases.